frontend/public/components/replicaset.jsx

```diff
@@ -21,10 +21,42 @@ import {
 
 import { ResourceEventStream } from './events';
 import { VolumesTable } from './volumes-table';
+import { k8sPatch } from '../module/k8s';
+import { confirmModal } from './modals';
 
 const {ModifyCount, AddStorage, common} = Kebab.factory;
 
-export const replicaSetMenuActions = [ModifyCount, AddStorage, ...common];
+const CancelAction = (kind, obj) => ({
+  label: 'Cancel Deployment',
+  hidden: !_.includes(['New', 'Pending', 'Running'], _.get(obj, ['metadata', 'annotations', 'openshift.io/deployment.phase'])),
+  callback: () => confirmModal({
+    title: 'Cancel deployment',
+    message: 'Are you sure you want to cancel this deployment?',
+    btnText: 'Yes, cancel',
+    cancelText: 'No, don\'t cancel',
+    executeFn: () => k8sPatch(kind,
+      obj,
+      [{
+        op: 'add',
+        path: '/metadata/annotations',
+        value: {
+          ...obj.metadata.annotations,
+          'openshift.io/deployment.cancelled': 'true',
+          'openshift.io/deployment.status-reason': 'cancelled by the user',
+        },
+      }]
```

Comment:
You shouldn't need to replace all of the annotations. You can instead just patch the values you need. `~1` can be used to escape `/`. (Untested, but I believe this should work.)

```js
[{
  op: 'add',
  path: '/metadata/annotations/openshift.io~1deployment.cancelled',
  value: 'true',
}, {
  op: 'add',
  path: '/metadata/annotations/openshift.io~1deployment.status-reason',
  value: 'cancelled by the user',
}]
```